infer/src/bo/arrayBlk.ml

```diff
@@ -0,0 +1,210 @@
+(*
+ * Copyright (c) 2016 - present
+ * Kihong Heo (http://ropas.snu.ac.kr/~khheo)
+ * Sungkeun Cho (http://ropas.snu.ac.kr/~skcho)
+ * Kwangkeun Yi (http://ropas.snu.ac.kr/~kwang)
+ *
+ * ROSAEC(Research On Software Analysis for Error-free Computing) Center
+ * Programming Research Laboratory
+ * Seoul National University, Korea
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *)
+
+(* Abstract Array Block *)
+open! IStd
+open BasicDom
+
+module ArrInfo =
+struct
+  type t =
+    { offset : Itv.t;
+      size : Itv.t;
+      stride : Itv.t; }
+
+  type astate = t
+
+  let bot : t
+  = { offset = Itv.bot; size = Itv.bot; stride = Itv.bot; }
+
+  let initial : t
+  = bot
+
+  let top : t
+  = { offset = Itv.top; size = Itv.top; stride = Itv.top; }
+
+  let input : t
+  = { offset = Itv.zero; size = Itv.pos; stride = Itv.one; }
+
+  let make : Itv.t * Itv.t * Itv.t -> t
+  = fun (o, s, stride) -> { offset = o; size = s; stride = stride; }
+
+  let compare : t -> t -> int
```

Comment:
In this case [https://github.com/ropas/infer/blob/inferbo/infer/src/bo/itv.ml#L830](url),
how can I use the annotation ```[@@deriving compare]```? 
The following code 
```ocaml
type t = astate [@@deriving compare]
```
yields 
```Error: Unbound value compare_astate```